validator/validator.go

```diff
@@ -0,0 +1,78 @@
+package validator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/multiformats/go-multiaddr"
+	"go.uber.org/zap/buffer"
+
+	addr "github.com/filecoin-project/go-address"
+)
+
+// NewValidatorFromString parses a string containing of validator address and multiaddress separated by "@".
+//
+// Example of validator strings:
+// - t1wpixt5mihkj75lfhrnaa6v56n27epvlgwparujy@/ip4/127.0.0.1/tcp/10000/p2p/12D3KooWJhKBXvytYgPCAaiRtiNLJNSFG5jreKDu2jiVpJetzvVJ
+// FIXME: Consider using json serde for this to support multiple multiaddr for validators.
+func NewValidatorFromString(s string) (Validator, error) {
+	parts := strings.Split(s, "@")
+	if len(parts) != 2 {
+		return Validator{}, fmt.Errorf("incorrect format of the string")
+	}
+
+	a, err := addr.NewFromString(parts[0])
+	if err != nil {
+		return Validator{}, err
+	}
+	m, err := multiaddr.NewMultiaddr(parts[1])
+	if err != nil {
+		return Validator{}, err
+	}
+
+	return Validator{
+		Addr:    a,
+		NetAddr: m.String(),
+	}, nil
+}
+
+type Validator struct {
+	Addr addr.Address `json:"addr"`
+	// FIXME: Consider using a multiaddr
+	NetAddr string `json:"net_addr,omitempty"`
+	Weight  uint64 `json:"weight"`
```

Comment:
Before merging let's discuss hoe this Weight is going to be used, it is not clear to me.

Reply:
Already synced on this, the field can stay, I will add the logic in the actor PR. We are just missing the comment about changing the type for the `ValidatorSet` in the state and feel free to merge.